### PR TITLE
MON-2801: ignore older release updates

### PR DESCRIPTION
**Pull request by rexagod**
merge only when upstream master version exceeds downstream.

### CODE REVIEW

File: .github/workflows/merge-flow.yaml [hunk 1]
```diff
@@ -77,9 +77,28 @@ jobs:
...
           echo "downstream=$(dirname ${{ inputs.downstream }})" >> $GITHUB_OUTPUT
           echo "sandbox=$(dirname ${{ inputs.sandbox }})" >> $GITHUB_OUTPUT
+          # assumes the presence of ./VERSION
+          DOWNSTREAM_VERSION=$(curl -sL "https://raw.githubusercontent.com/${{ inputs.downstream }}/${{ inputs.downstream-branch }}/VERSION")
+          echo "downstream-version=${DOWNSTREAM_VERSION}" >> $GITHUB_OUTPUT
+      - uses: madhead/semver-utils@latest
+        id: version
+        with:
+          version: ${{ steps.org.outputs.downstream-version }}
+          compare-to: ${{ steps.upstream.outputs.release }}
```
Comment (rexagod) [inline .github/workflows/merge-flow.yaml:87]: [`vx.y.z` and `x.y.z` are equivalent.](https://github.com/npm/node-semver/blob/cce61804ba6f997225a1267135c06676fe0524d2/functions/clean.js#L3)

File: .github/workflows/merge-flow.yaml [hunk 2]
```diff
@@ -77,9 +77,28 @@ jobs:
           echo "upstream=$(dirname ${{ inputs.upstream }})" >> $GITHUB_OUTPUT
           echo "downstream=$(dirname ${{ inputs.downstream }})" >> $GITHUB_OUTPUT
           echo "sandbox=$(dirname ${{ inputs.sandbox }})" >> $GITHUB_OUTPUT
+          # assumes the presence of ./VERSION
+          DOWNSTREAM_VERSION=$(curl -sL "https://raw.githubusercontent.com/${{ inputs.downstream }}/${{ inputs.downstream-branch }}/VERSION")
+          echo "downstream-version=${DOWNSTREAM_VERSION}" >> $GITHUB_OUTPUT
+      - uses: madhead/semver-utils@latest
+        id: version
+        with:
```
Comment (simonpasquier) [inline .github/workflows/merge-flow.yaml:85]: could we set `lenient: false` to ensure that the action fails in case it can't parse the version?

**Review by simonpasquier — COMMENTED** [review]
one small ask otherwise lgtm

**Review by simonpasquier — APPROVED** [review]
lgtm

**Pull request merged**